File: src/main/java/hilfling/backend/hilfling/model/ArticleTag.java
package hilfling.backend.hilfling.model;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;

@Data
@Entity(name = "article_tags")
@Table(name = "article_tags")
public class ArticleTag implements Serializable, BaseModel {
    public ArticleTag() {
    }

    public ArticleTag(String tag) {
        this.tag = tag;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "tag", nullable = false, unique = true)
    private String tag;
}

File: src/main/kotlin/hilfling/backend/hilfling/model/ArticleTag.kt
package hilfling.backend.hilfling.model

import me.liuwj.ktorm.database.Database
import me.liuwj.ktorm.entity.Entity
import me.liuwj.ktorm.entity.sequenceOf
import me.liuwj.ktorm.schema.*

interface ArticleTag : BaseModel<ArticleTag> {
    companion object : Entity.Factory<ArticleTag>()

    var name: String
}

object ArticleTags : BaseTable<ArticleTag>("article_tag") {
    val name = varchar("name").bindTo { it.name }
}

val Database.article_tags get() = this.sequenceOf(ArticleTags)